rasa/utils/tensorflow/model_data.py

```diff
@@ -19,10 +20,22 @@
 import scipy.sparse
 from sklearn.model_selection import train_test_split
 
-
 logger = logging.getLogger(__name__)
 
 
+def ragged_array_to_ndarray(ragged_array: Iterable[np.ndarray]) -> np.ndarray:
+    """Converts list (iterable) of numpy arrays to numpy array.
+
+    Try to convert as is (preserves type), if fails becuase not all numpy arrays has
+    the same shape, then creates numpy array of objects.
+    """
+    try:
+        x = np.array(ragged_array)
+    except ValueError:
+        x = np.array(ragged_array, dtype=object)
```

Comment:
I would return directly in each line (L33 or L35) or use a more descriptive name `converted_np_array` or something along these lines.

Reply:
One thing I forgot to add: please add a unit test for this helper function (checking both scenarios)